### PR TITLE
Move NavViewItem presenter margin to resource

### LINKED ISSUE #2288: Move NavigationViewItemPresenter ContentPresenter margin to a resource
As of now, the "ContentPresenter" margin in `NavigationViewItemPresenter` is being defined in the control template. Move it to a resource so that it can be light weight styled as mentioned in [this comment](https://github.com/microsoft/microsoft-ui-xaml/pull/2280#discussion_r407627637).

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

Move NavigationViewItem's ContentPresenter margin to resource for lightweight styling

In the current state, this PR does not create merge conflicts with #2271 
## Motivation and Context

Closes #2288

## How Has This Been Tested?

Running app and looking at visuals. 
## Screenshots (if appropriate):

Open question: Should the negative margin of the chevron also be a theme resource?

### CODE REVIEW

**Comment by Felix-Dev** [comment]
> Should the negative margin of the chevron also be a theme resource?

@chingucoding I started a discussion the chevron [here](https://github.com/microsoft/microsoft-ui-xaml/issues/2287).

**Comment by chingucoding** [comment]
> @chingucoding I started a discussion about the chevron here.

I think that discussion is more moving towards changing that value, right? The question in this PR is to what degree we should provide lightweight styling.

**Comment by Felix-Dev** [comment]
@Chingucoding Since users can lightweight style the thickness of the scrollbar and the linked issue is meant to prevent the chevron from being overlapped by the scrollbar we should also provide a lighweight styling resource for the chevron.

**Comment by chingucoding** [comment]
True. @ranjeshj @ojhad @StephenLPeters @chigy Thoughts?

**Comment by Felix-Dev** [comment]
@Chingucoding How do we want to approach issues #2288 and #2287? My initial plan was to create one PR with the updated values *and* adding lightweight styling resources. Now that you already created a PR for this one, do we consolidate them in one PR or wait for one PR to be merged first with master and then base the other PR on it?

**Comment by chingucoding** [comment]
If that's fine with you, I can add the fix for #2288 it in this PR to save time with this, but I leave that up to you. If you want you can create a PR with the updated values, you can also do that. Since we wouldn't be touching the same areas depending on where you put the new resource, I don't think that there will be merge issues.

**Comment by Felix-Dev** [comment]
Talked with @chingucoding and we decided to treat both issues independently. A later PR for issue #2287 will update the introduced lighweight styling resources in this PR if required.

**Comment by StephenLPeters** [comment]
>                                     Margin="-16,0,0,0"

We'd want this to update with changes to the resource value as well

---
Refers to: dev/NavigationView/NavigationView_rs1_themeresources.xaml:677 in e6e3cd0. [](commit_id = e6e3cd0db2da35cead70f13aa141e0a9727b0b36, deletion_comment = False)

**Comment by StephenLPeters** [comment]
> > ```
> >                                 Margin="-16,0,0,0"
> > ```
> 
> We'd want this to update with changes to the resource value as well
> 
> Refers to: dev/NavigationView/NavigationView_rs1_themeresources.xaml:677 in [e6e3cd0](https://github.com/microsoft/microsoft-ui-xaml/commit/e6e3cd0db2da35cead70f13aa141e0a9727b0b36). [](commit_id = [e6e3cd0](https://github.com/microsoft/microsoft-ui-xaml/commit/e6e3cd0db2da35cead70f13aa141e0a9727b0b36), deletion_comment = False)

I'm not sure there is a way to get both of these referencing the same resource without creating a converter... So I think I'd just make a second resource

**Comment by chingucoding** [comment]
I've added a resource for that. Also noticed that we had some special margins for the top present, added resources for them too.

**Comment by ranjeshj** [comment]
@chingucoding #2271 is now merged, can you merge master and push an update ? Thanks!

**Comment by chingucoding** [comment]
> @chingucoding #2271 is now merged, can you merge master and push an update ? Thanks!

Good idea, done.

**Comment by ojhad** [comment]
>                                     Margin="12,0,16,0"

there is a separate style for when these items are in the overflow flyout, we probably want to create styles for these as well

---
Refers to: dev/NavigationView/NavigationView_rs1_themeresources.xaml:1468 in c7d4a15. [](commit_id = c7d4a1508ebbc41fe89b037fdb8f5295caf0f465, deletion_comment = False)

**Comment by chingucoding** [comment]
> there is a separate style for when these items are in the overflow flyout, we probably want to create resources for these as well

Good point, created resources for that too.

File: dev/NavigationView/NavigationView_rs1_themeresources.xaml [hunk 1]
```diff
@@ -248,7 +248,13 @@
     <Thickness x:Key="NavigationViewItemSeparatorMargin">16,10</Thickness>
     <Thickness x:Key="NavigationViewCompactItemSeparatorMargin">16,10</Thickness>
     <Thickness x:Key="TopNavigationViewItemSeparatorMargin">10,0</Thickness>
-
+    <Thickness x:Key="NavigationViewItemContentPresenterMargin">0,0,16,0</Thickness>
+    <Thickness x:Key="TopNavigationViewItemContentPresenterMargin">8,0,16,0</Thickness>
+    <Thickness x:Key="NavigationViewItemExpandChevronMargin">-16,0,0,0</Thickness>
+    <Thickness x:Key="TopNavigationViewItemExpandChevronMargin">-16,0,0,0</Thickness>
+    <Thickness x:Key="TopNavigationViewItemOnOverflowExpandPresenterMargin">12,0,16,0</Thickness>
```
Comment (ojhad) [inline dev/NavigationView/NavigationView_rs1_themeresources.xaml:255]: I think you mean `TopNavigationViewItemOnOverflowContentPresenterMargin`

**Pull request merged**